pkg/apis/crd/v1alpha1/types.go

```diff
@@ -354,3 +355,144 @@ type BGPPeer struct {
 	// a restart before deleting stale routes. The range of the value is from 1 to 3600, and the default value is 120.
 	GracefulRestartTimeSeconds *int32 `json:"gracefulRestartTimeSeconds,omitempty"`
 }
+
+type PodReference struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+type ServiceReference struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+// Source describes the source spec of the packetcapture.
+type Source struct {
+	// Pod is the source Pod.
+	Pod *PodReference `json:"pod,omitempty"`
+	// IP is the source IPv4 or IPv6 address.
+	IP *string `json:"ip,omitempty"`
+}
+
+// Destination describes the destination spec of the PacketCapture.
+type Destination struct {
+	// Pod is the destination Pod, exclusive with destination Service.
+	Pod *PodReference `json:"pod,omitempty"`
+	// Service is the destination Service, exclusive with destination Pod.
+	Service *ServiceReference `json:"service,omitempty"`
+	// IP is the destination IPv4 or IPv6 address.
+	IP *string `json:"ip,omitempty"`
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	UDP *UDPHeader `json:"udp,omitempty"`
+	TCP *TCPHeader `json:"tcp,omitempty"`
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	// SrcPort is the source port.
+	SrcPort *int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort *int32 `json:"dstPort,omitempty"`
+	// Flags are flags in the header.
+	Flags *int32 `json:"flags,omitempty"`
+}
+
+// Packet includes header info.
+type Packet struct {
+	// IPFamily is the filter's IP family. Default to `IPv4`.
+	IPFamily v1.IPFamily `json:"ipFamily,omitempty"`
+	// Protocol represents the transport protocol. Default is to not filter on protocol.
+	Protocol        *intstr.IntOrString `json:"protocol,omitempty"`
+	TransportHeader TransportHeader     `json:"transportHeader"`
+}
+
+// PacketCaptureFirstNConfig contains the config for the FirstN type capture. The only supported parameter is
+// `Number` at the moment, meaning capturing the first specified number of packets in a flow.
+type PacketCaptureFirstNConfig struct {
+	Number int32 `json:"number"`
+}
+
+const DefaultPacketCaptureTimeout uint16 = 60
```

Comment:
is this used anywhere?

Reply:
in the datapath code, or we can move it there.

Reply:
removed.